_posts/2021-09-23-reassign-partitions.md

```diff
@@ -0,0 +1,282 @@
+  replicas: 3
+  config:
+    retention.ms: 7200000
+    segment.bytes: 1073741824
+    # ...
+```
+
+Once the cluster is running, we will need a user.
+We will configure the user for TLS client authentication and configure authorization to allow it to send and receive examples from a topics named `my-topic` and 
+`my-topic-1`:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaUser
+metadata:
+  name: my-user
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+  authentication:
+    type: tls
+  authorization:
+    type: simple
+    acls:
+      - resource:
+          type: topic
+          name: my-topic
+          patternType: literal
+        operation: Write
+        host: "*"
+      - resource:
+          type: topic
+          name: my-topic-1
+          patternType: literal
+        operation: Read
+        host: "*"
+      - resource:
+          type: topic
+          name: my-topic
+          patternType: literal
+        operation: Describe
+        host: "*"        
+      - resource:
+          type: topic
+          name: my-topic-1
+          patternType: literal
+        operation: Describe
+        host: "*"
+  # ...
+```
+
+### Procedure
+
+Once you have a running cluster with topics and the brokers, now you are ready to create the proposal json data.
+The fist step would be to extract the certificates and key for the cluster and the user in order to authenticate using Tls.
+
+Let us proceed toward extracting the cluster CA certificate and password.
+We can extract both of them easily from the `<CLUSTER-NAME>-cluster-ca-cert` Secret (where `<CLUSTER-NAME>` denotes the name of the cluster) using the following commands:
+
+```sh
+kubectl get secret CLUSTER-NAME-cluster-ca-cert -o jsonpath='{.data.ca\.p12}' | base64 -d > ca.p12
+```
+
+```sh
+kubectl get secret CLUSTER-NAME-cluster-ca-cert -o jsonpath='{.data.ca\.password}' | base64 -d > ca.password
+```
+
+Before proceeding toward extracting the user credentials, create a separate interactive pod.
+This interactive pod will be used to run all the reassignment commands. 
+One question might bug you. What is the need of a separate pod? Can't We use one of the broker pods?
+The answer to this is running commands from within a broker is not a good practice and is insecure. So it is always better to avoid running the command from a broker pod.
+
+So now it's time to get our interactive pod up and running. You can use the following command:
+
+```sh
+kubectl run --restart=Never --image=quay.io/strimzi/kafka:latest-kafka-2.8.0 <interactive_pod_name> -- /bin/sh -c "sleep 3600"
+```
+
+Wait till the pod gets into the `Ready` state. Once the pod gets into `Ready` state, now our next step will be to copy the CA certificate to the interactive pod container:
+
+```sh
+kubectl cp ca.p12 <interactive_pod_name>:/tmp
+```
+
+We can now proceed to fetch the CA certificate and password for the user. User CA certificate and password are extracted from the User Secret like this :
```

Comment:
It is not CA certifciate here. It is only user certificate in both cases here.

Reply:
How can I get/create these, if I have 0 kafka users? I am unable to find any how-to on TLS users.